broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/PartitionTransitionException.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.system.partitions.impl;
+
+/**
+ * This exception should be used if the installation of the partition service's fail, and if the
+ * failure should not be handled.
+ */
+public class PartitionTransitionException extends RuntimeException {
```

Comment:
Suggestion: Rename it to `PartitionTransitionAbortedException` to indicate that the transition was aborted intentionally and should not treat it as a failure. Alternative `RecoverablePartitionTransitionException`. Wdyt?

Reply:
That's great, @deepthidevaki. I will go with `RecoverablePartitionTransitionException`, to me, it is pretty on point. Thank you!

Reply:
Maybe it makes sense to extend https://github.com/camunda-cloud/zeebe/blob/main/util/src/main/java/io/camunda/zeebe/util/exception/RecoverableException.java ? :thinking:

Reply:
Thanks, @Zelldon. If I am not mistaken, then you mean that this newly introduced exception class could extend from `RecoverableException`. I also considered this when introducing that new exception, but decided against it.

I thought of keeping it "local" to the `PartitionTransition` concept. Additionally, to my understanding, when going through the usage of the `RecoverableException`, it will be handled by triggering some kind of "action", for instance, the stream processor will retry to process the command:

https://github.com/camunda-cloud/zeebe/blob/52d2be30cdc9245ca15917303e97807b58ed682b/engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ProcessingStateMachine.java#L257-L264

This differs from this new exception class. In this case, we want to indicate that the transition was aborted intentionally, should not be handled as a failure and basically, drop it because it gets resolved on its own. That's why I decided to not extend from `RecoverableException`.